test/common/http/path_utility_test.cc

```diff
@@ -133,5 +215,39 @@ TEST_F(PathUtilityTest, RemoveQueryAndFragment) {
   EXPECT_EQ("/abc", PathUtil::removeQueryAndFragment("/abc?param=value#fragment"));
 }
 
+class PathTransformerTest : public testing::Test {
+public:
+  void setPathTransformer(std::vector<std::string> transformations_config) {
+    envoy::type::http::v3::PathTransformation path_transformation_config;
+    Protobuf::RepeatedPtrField<envoy::type::http::v3::PathTransformation_Operation>* operations =
+        path_transformation_config.mutable_operations();
+    for (std::string const& transformation : transformations_config) {
+      auto* operation = operations->Add();
+      if (transformation == "NormalizePathRFC3986") {
+        operation->mutable_normalize_path_rfc_3986();
+      } else if (transformation == "MergeSlashes") {
+        operation->mutable_merge_slashes();
+      }
+    }
+    path_transformer_.reset(new PathTransformer(path_transformation_config));
+  }
+  const PathTransformer& pathTransformer() { return *path_transformer_; }
+
+  std::unique_ptr<PathTransformer> path_transformer_;
+};
+
+TEST_F(PathTransformerTest, MergeSlashes) {
+  setPathTransformer({"MergeSlashes"});
+  PathTransformer const& path_transformer = pathTransformer();
+  EXPECT_EQ("", path_transformer.transform("").value());                // empty
+  EXPECT_EQ("a/b/c", path_transformer.transform("a//b/c").value());     // relative
+  EXPECT_EQ("/a/b/c/", path_transformer.transform("/a//b/c/").value()); // ends with slash
+  EXPECT_EQ("a/b/c/", path_transformer.transform("a//b/c/").value());   // relative ends with slash
+}
+
+TEST_F(PathTransformerTest, DuplicateTransformation) {
```

Comment:
great test! Could you also add one duplicated but not adjacent?

Reply:
Thanks, will do.